python/spark_sklearn/tests/test_gapply.py

```diff
@@ -0,0 +1,211 @@
+
+from itertools import chain
+import pandas as pd
+import random
+import unittest
+
+from pyspark.conf import SparkConf
+from pyspark.sql import SparkSession
+from pyspark.sql.types import *
+from pyspark.sql import Row
+from pyspark.sql.tests import PythonOnlyPoint, PythonOnlyUDT, ExamplePoint, ExamplePointUDT
+
+from spark_sklearn.test_utils import fixtureReuseSparkSession
+from spark_sklearn.group_apply import gapply
+
+def _assert_frame_equal(actual, expected):
+    # Points are unhashable, so pandas' assert_frame_equal can't check they're the same by default.
+    # We need to convert them to something that can be checked.
+    def convert_points(pt):
+        if any(isinstance(pt, t) for t in (PythonOnlyPoint, ExamplePoint)):
+            return (pt.x, pt.y)
+        return pt
+    def normalize(df):
+        converted = df.apply(lambda col: col.apply(convert_points))
```

Comment:
I can't think of a comment that wouldn't be redundant with the code (bc variable names already indicate it) - what do you want me to say?